librad/src/git/storage.rs

```diff
@@ -145,7 +150,9 @@ pub enum RadSelfSpec {
 }
 
 pub type NoSigner = PhantomData<!>;
-pub type WithSigner = SecretKey;
+
+pub trait WithSigner: sign::Signer + Send + Sync + Clone + 'static {}
+impl WithSigner for SecretKey {}
```

Comment:
To make the `WithSigner` trait usable out of the box with other `keystore` Signer implementations I’d suggest the following
```suggestion
impl<T: sign::Signer + Send + Sync + Clone + 'static> WithSigner for T {}
```

Reply:
Ah yes! That was the missing piece I forgot. Thank you :grin: